### PR TITLE
Auto-reset: do not trigger/process reset task when reset point has e…

### PR DESCRIPTION
**Pull request by longquanzheng**
…xpired

### CODE REVIEW

File: service/history/workflowResetor_test.go [hunk 1]
```diff
@@ -4204,4 +4219,31 @@ func TestFindAutoResetPoint(t *testing.T) {
...
+		Binaries: map[string]*workflow.BadBinaryInfo{
+			"none":    {},
+			"expired": {},
+		},
+	}, &workflow.ResetPoints{
+		Points: []*workflow.ResetPointInfo{
+			pt0, pt1, pt3, pt4, pt5,
+		},
+	})
+	fmt.Println(pt.GetBinaryChecksum())
```
Comment (shreyassrivatsan) [inline service/history/workflowResetor_test.go]: spurious print?

File: service/history/workflowResetor.go [hunk 2]
```diff
@@ -860,9 +860,16 @@ func FindAutoResetPoint(badBinaries *workflow.BadBinaries, autoResetPoints *work
 	if badBinaries == nil || badBinaries.Binaries == nil || autoResetPoints == nil || autoResetPoints.Points == nil {
 		return
 	}
+	nowNano := time.Now().UnixNano()
 	for _, p := range autoResetPoints.Points {
 		bin, ok := badBinaries.Binaries[p.GetBinaryChecksum()]
 		if ok && p.GetResettable() {
+			if p.GetExpiringTimeNano() > 0 {
+				if nowNano > p.GetExpiringTimeNano() {
```
Comment (shreyassrivatsan) [inline service/history/workflowResetor.go]: nit: combine these conditions into a single if statement

**Comment by shreyassrivatsan** [comment]
Approved with a couple of comments

File: service/history/workflowResetor.go (hunk 2 above)
Comment (longquanzheng) [reply service/history/workflowResetor.go]: Sure

File: service/history/workflowResetor_test.go (hunk 1 above)
Comment (longquanzheng) [reply service/history/workflowResetor_test.go]: Oh thanks for pointing it out.

**Pull request merged**